### PR TITLE
Add tests for copy construction and assignment

### PR DESCRIPTION
**Pull request by Twon**

### CODE REVIEW

File: test_indirect.cpp [hunk 1]
```diff
@@ -12,6 +12,70 @@ TEST_CASE("Defeault construction for indirect", "[constructor.default]")
...
+                REQUIRE(*copy_of_a == a_value); 
+                REQUIRE(a.operator->() != nullptr); 
+                REQUIRE(copy_of_a.operator->() != nullptr); 
+                REQUIRE(a.operator->() != copy_of_a.operator->()); 
+            }
+        }
+    }
+}
+
+TEST_CASE("Copy assignment for indirect of a primative type", "[assignment.copy.primitive]")
```
Comment (jbcoe) [inline test_indirect.cpp]: Typo: primitive

File: test_indirect.cpp [hunk 2]
```diff
@@ -12,6 +12,70 @@ TEST_CASE("Defeault construction for indirect", "[constructor.default]")
...
+{
+    GIVEN("A value initialised indirect value")
```
Comment (jbcoe) [inline test_indirect.cpp]: I would hyphenate ‘value-initialised’ but am happy without if you feel strongly.

File: test_indirect.cpp [hunk 3]
```diff
@@ -12,6 +12,70 @@ TEST_CASE("Defeault construction for indirect", "[constructor.default]")
...
+        {
+             constexpr int b_value = 10;
+             indirect<int> b{ new int(b_value) };
+             REQUIRE(*b == b_value);
+             //int const * const location_of_b = b.operator->();
+
+             THEN("The assigned too object make a deep copy of the original value without futher unnecessary allocation")
+             {
+                 b = a;
+                 // REQUIRE(location_of_b == b.operator->()); Should we ensure that no allocation happens, or is the impl defined?
```
Comment (jbcoe) [inline test_indirect.cpp]: we should take ownership so no allocation or copying can occur

File: test_indirect.cpp [hunk 4]
```diff
@@ -12,6 +12,70 @@ TEST_CASE("Defeault construction for indirect", "[constructor.default]")
...
+             }
+        }
+        WHEN("Assigning a copy into a value initalised indirect value")
+        {
+             constexpr int b_value = 10;
+             indirect<int> b{ new int(b_value) };
+             REQUIRE(*b == b_value);
+             //int const * const location_of_b = b.operator->();
+
+             THEN("The assigned too object make a deep copy of the original value without futher unnecessary allocation")
```
Comment (jbcoe) [inline test_indirect.cpp]: Typo: “assigned to”

File: README.md [hunk 5]
```diff
@@ -4,3 +4,21 @@
 [![GitHub Issues](https://img.shields.io/github/issues/jbcoe/indirect.svg)](http://github.com/jbcoe/indirect/issues)
 
 # indirect
+
+- Constant propagation from owner to contained value type
+- Deep copy semantics
+-
+
+# License
```
[the remark this answers is not in the document]
Comment (Twon) [reply README.md]: Yes, it unfortunate that you can't include text from other files in a markdown file.  Cool, I've removed.

File: test_indirect.cpp (hunk 2 above)
Comment (Twon) [reply test_indirect.cpp]: No, I agree, I just didn't think about it.  Corrected all instance of it.

File: test_indirect.cpp (hunk 1 above)
Comment (Twon) [reply test_indirect.cpp]: Good spot

File: test_indirect.cpp (hunk 4 above)
Comment (Twon) [reply test_indirect.cpp]: Corrected

**Pull request merged**